Add a warning to the docs that TFC Agent doesn't support TFE_PARALLELISM (#1772)
hashicorp/terraform-website: content/source/docs/cloud/workspaces/variables.html.md

- `TFE_PARALLELISM` — If present, Terraform Cloud uses this to set `terraform plan` and `terraform apply`'s `-parallelism=<N>` flag ([more info](/docs/internals/graph.html#walking-the-graph)). Valid values are between 1 and 256, inclusive; the default is `10`. This is rarely necessary, but can fix problems with infrastructure providers that error on concurrent operations or use non-standard rate limiting. We recommend talking to HashiCorp support before using this.

> Please note that `TFE_PARALLELISM` is not supported by Terraform Cloud Agents. Modern versions of Terraform allow you to specify flags as environment variables directly via [TF_CLI_ARGS](/docs/cli/config/environment-variables.html#tf-cli-args). Use `TF_CLI_ARGS="parallelism=<N>"` instead.

Reviewer: Adding the note callout notation which will make this appear as a blue note box! I recommend doing this so this stands out. Also - when we say "modern versions" should we specify a version number to be more specific?

```suggestion
-> **Note:** `TFE_PARALLELISM` is not supported by Terraform Cloud Agents, but modern versions of Terraform allow you to specify flags as environment variables directly via [TF_CLI_ARGS](/docs/cli/config/environment-variables.html#tf-cli-args). Use `TF_CLI_ARGS="parallelism=<N>"` instead.
```

Author: This environment variable is quite old. I believe it was [introduced in 0.9.0](https://github.com/hashicorp/terraform/blob/b0711a0585c44f6ebb1a7c7ef8bedd5de8ac17ac/CHANGELOG.md#L1405). My intention with "modern" was just to explain why `TFE_PARALLELISM` ever existed. I'm fine with being specific, but TFC doesn't actually support anything older than 0.9.0.
